internal/extract.go

```diff
@@ -55,6 +58,36 @@ func (err UnsupportedFileTypeError) Error() string {
 	return fmt.Sprintf(tracelog.GetErrorFormatter(), err.error)
 }
 
+type FileExtractor interface {
```

Comment:
I see some existing logic duplication there. Why we can't use the regular `TarInterpreter`? As I can see, the main difference is only to append the relative path to the destination file path. I guess that we can modify the `tar.Header` and insert the relative path there instead of introducing the new `FileExtractor`.

Reply:
Remove `FileExtractor` and use `tarInterpreter.Interpret(..., &tar.Header{...})` in raw files. Also add some info in `ReaderMarker` to understand: is that a tar file or regular file